test/lit/passes/memory-packing-gc.wast

```diff
@@ -0,0 +1,166 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; Test that memory-packing does the right thing in the presence of array.new_data.
+
+;; RUN: foreach %s %t wasm-opt --memory-packing --all-features -S -o - | filecheck %s
+
+(module
```

Comment:
Please add brief explanations for the testcases, e.g. here `array.new_data refers to this data segment so it cannot be removed`.